packages/flutter_tools/lib/src/bundle.dart

```diff
@@ -218,17 +219,29 @@ void assemble({
   printTrace('Wrote $assetDirPath');
 }
 
-void writeBundle(
+Future<void> writeBundle(
   Directory bundleDir,
   Map<String, DevFSContent> assetEntries,
-) {
-  if (bundleDir.existsSync())
+) async {
+  if (bundleDir.existsSync()) {
     bundleDir.deleteSync(recursive: true);
+  }
   bundleDir.createSync(recursive: true);
+  // macOS has limited file descriptors, so attempting to copy all files using
+  // the async methods has a risk of exhausting them. Instead we use a pool to
+  // limit ourselves to 100 at a time.
+  final Pool pool = Pool(100);
```

Comment:
Could we make this use the actual amount of file descriptors available?  or perhaps half of them or something?  e.g. use the output of `ulimit -n` on Mac/Linux, not sure what that would be for Windows.

Reply:
e.g. a new method in os.dart or something like how we do `which`

Reply:
I sort of remember that ulimit on Mac can lie, like you can set it to a bigger number of fd's, but that it is not ground truth. I would suggest hardcoding this number down even further, like to 32 or so. The Dart VM's IO service thread isn't going to be able to service more concurrent operations than that anyway.

Reply:
That seems fair to me - then we could document that this is the max number the IO service can handle at once?